src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/RSAPssX509SignatureGeneratorTests.cs

```diff
@@ -0,0 +1,177 @@
+                Assert.Equal(new byte[] { 5, 0 }, publicKey.EncodedParameters.RawData);
+
+                string expectedKeyHex =
+                    // SEQUENCE
+                    "3082010C" +
+                    //   INTEGER (modulus)
+                    "0282010100" + TestData.RsaBigExponentParams.Modulus.ByteArrayToHex() +
+                    //   INTEGER (exponent)
+                    "0205" + TestData.RsaBigExponentParams.Exponent.ByteArrayToHex();
+
+                Assert.Equal(expectedKeyHex, publicKey.EncodedKeyValue.RawData.ByteArrayToHex());
+
+                const string rsaEncryptionOid = "1.2.840.113549.1.1.1";
+                Assert.Equal(rsaEncryptionOid, publicKey.Oid.Value);
+                Assert.Equal(rsaEncryptionOid, publicKey.EncodedParameters.Oid.Value);
+                Assert.Equal(rsaEncryptionOid, publicKey.EncodedKeyValue.Oid.Value);
+
+                PublicKey publicKey2 = signatureGenerator.PublicKey;
+                Assert.Same(publicKey, publicKey2);
+            }
+        }
+
+        [Theory]
+        [InlineData("SHA256")]
+        [InlineData("SHA384")]
+        [InlineData("SHA512")]
+        public static void SignatureAlgorithm_StableNotSame(string hashAlgorithmName)
+        {
+            using (RSA rsa = RSA.Create())
+            {
+                RSAParameters parameters = TestData.RsaBigExponentParams;
+                rsa.ImportParameters(parameters);
+
+                var signatureGenerator = X509SignatureGenerator.CreateForRSA(rsa, RSASignaturePadding.Pss);
+
+                HashAlgorithmName hashAlgorithm = new HashAlgorithmName(hashAlgorithmName);
+
+                byte[] sigAlg = signatureGenerator.GetSignatureAlgorithmIdentifier(hashAlgorithm);
+                byte[] sigAlg2 = signatureGenerator.GetSignatureAlgorithmIdentifier(hashAlgorithm);
+
+                Assert.NotSame(sigAlg, sigAlg2);
+                Assert.Equal(sigAlg, sigAlg2);
+            }
+        }
+
+        [Theory]
+        [InlineData("MD5")]
+        [InlineData("SHA1")]
+        [InlineData("Potato")]
+        public static void SignatureAlgorithm_NotSupported(string hashAlgorithmName)
+        {
+            using (RSA rsa = RSA.Create())
+            {
+                RSAParameters parameters = TestData.RsaBigExponentParams;
+                rsa.ImportParameters(parameters);
+
+                var signatureGenerator = X509SignatureGenerator.CreateForRSA(rsa, RSASignaturePadding.Pss);
+
+                HashAlgorithmName hashAlgorithm = new HashAlgorithmName(hashAlgorithmName);
+
+                Assert.Throws<ArgumentOutOfRangeException>(
+                    "hashAlgorithm",
+                    () => signatureGenerator.GetSignatureAlgorithmIdentifier(hashAlgorithm));
+            }
+        }
+
+        [Theory]
+        [InlineData("SHA256")]
+        [InlineData("SHA384")]
+        [InlineData("SHA512")]
+        public static void SignatureAlgorithm_Encoding(string hashAlgorithmName)
```

Comment:
Is there a way to compare to something generated by another stack? That would help validate that these are, in fact, the right bytes to be generating.